src/script_interface/Variant.hpp

```diff
@@ -29,6 +31,64 @@ enum class VariantType {
   VECTOR
 };
 
+namespace detail {
+template <typename T, typename = void> struct infer_type_helper {};
+
+template <> struct infer_type_helper<bool> {
+  static constexpr VariantType value{VariantType::BOOL};
+};
+
+template <> struct infer_type_helper<std::string> {
+  static constexpr VariantType value{VariantType::STRING};
+};
+
+template <typename T>
+struct infer_type_helper<
+    T, typename std::enable_if<std::is_integral<T>::value, void>::type> {
+  static constexpr VariantType value{VariantType::INT};
+};
```

Comment:
This substitution allows down-conversion from `int` to `short` or even `char`.  I wouldn't make the assumption what people only pass sufficiently short values through this.